### PR TITLE
Feature/82 extend usage of context

### PR DESCRIPTION
**Pull request by timohuber**
I added the following user to the context:

```
type user =
  | Admin of Sihl_user.t
  | Contact of Contact.t
  | Root of Sihl_user.t
```

I used `Sihl_user.t` as a placeholder for `Admin.t` until `guardian` is implemented.

### CODE REVIEW

File: pool/web/view/page/page_layout.ml [hunk 1]
```diff
@@ -130,33 +137,51 @@ module Tenant = struct
...
+    language
+    query_language
+    active_navigation
+    tenant_languages
+    active_lang
+    =
+    let open Pool_common.I18n in
+    let open Pool_context in
+    let to_main_nav = to_main_nav language query_language active_navigation in
+    let language_switch () = i18n_links tenant_languages active_lang in
```
Comment (aronerben) [inline pool/web/view/page/page_layout.ml:151]: Does this need the `()` arg?

**Review by aronerben — CHANGES_REQUESTED** [review]

File: pool/web/view/page/page_layout.ml (hunk 1 above)
Comment (timohuber) [reply pool/web/view/page/page_layout.ml:151]: No, and I also realised that I added it to the wrong navigation.

**Comment by timohuber** [comment]
@aronerben ping

File: pool/app/pool_common/entity.ml [hunk 2]
```diff
@@ -274,3 +274,10 @@ module ExperimentType = struct
       PoolError.Field.ExperimentType
   ;;
 end
+
+module Sihl_user = struct
```
Comment (aronerben) [inline pool/app/pool_common/entity.ml:278]: I have an idea, what if we override this in `service.ml` itself? That way we can ensure that whoever uses `Service.User` uses this version

**Comment by aronerben** [comment]
There are merge conflicts @timohuber

**Review by aronerben — APPROVED** [review]

**Pull request merged**